.github/workflows/rust.yml

```diff
@@ -781,3 +781,11 @@ jobs:
       - name: Check datafusion-proto
         working-directory: datafusion/proto
         run: cargo msrv --output-format json --log-target stdout verify
+  typos:
+    name: Spell Check with Typos
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+        with:
+          persist-credentials: false
+      - uses: crate-ci/typos@v1
```

Comment:
Per https://infra.apache.org/github-actions-policy.html this should use commit hash.
I am not sure https://github.com/apache/infrastructure-actions/blob/main/approved_patterns.yml is normative. In any case, i read it as saying we can use the action. Not saying that we can safely use the action without pinning to a hash.